### ISSUE TITLE
gridfield urgently needs UX attention

### ISSUE THREAD

**Issue by sunnysideup** [silverstripe/silverstripe-framework#8715]
## Affected Version

SS4.3

## Description

Every time my colleague says that wordpress has a better gridfield I am upset.  Here is an example of a basic gridfield.  It gives me the impression that Wordpress is not even that great. Our gridfield looks so disconnected.  A bunch of elements thrown on a page without any cohesion. I would love to SIlverstripe to be the best CMS in the world, but with this gridfield (the KEY element of the CMS), this will not happen.  This is in sharp contrast with some of the brilliance in other areas which gives me the impression that UX does not get enough attention. This is the shame, because the whole purpose of a CMS is to make data-entry easier. 

![image](https://user-images.githubusercontent.com/167154/50983827-382f1280-1565-11e9-9f1f-bfc1516b1e60.png)

Basically it boils down to two basic issues here:

(a) someone needs to redesign the header, make it tighter, let the design be determined by its function, etc...
(b) we need margins around on the left and right of the rows.

**Comment by robbieaverill** [comment]
Hey @sunnysideup 

As a side note - you can remove the "View 1-5 of 5" page count component and the toolbar it sits in, if you'd like to remove that mostly empty space from your GridField:

```php
$gridField->getConfig()->removeComponentsByType([
    GridFieldPageCount::class,
    GridFieldToolbarHeader::class,
]);
```

We are working on a new React/GraphQL driven GridField which will make its way eventually into the SilverStripe 4.x release line - see https://github.com/silverstripe/silverstripe-admin/issues/556 https://github.com/silverstripe/silverstripe-admin/issues/537 https://github.com/silverstripe/silverstripe-admin/issues/463 for reference =)

**Comment by sunnysideup** [comment]
ahhhh that is such good news!  Thank you.  The thing is - and this is so important to understand.  We are upgrading sites to 4.0 and then when the gridfield looks so messy you are going to have a lot of disappointed clients.

**Comment by robbieaverill** [comment]
@sunnysideup I've also raised an enhancement to improve the design of the add existing autocomplete component: https://github.com/silverstripe/silverstripe-admin/issues/800

**Comment by dnsl48** [comment]
@sunnysideup thank you for the feedback!
I can see you already commented on the React GridField issue https://github.com/silverstripe/silverstripe-admin/issues/463 and we would love to hear more ideas from you in there.
However, to be more productive, we don't want to spread the discussion across too many github issues, so I'm closing this one.

**Comment by robbieaverill** [comment]
Hah, here's a duplicate from @sunnysideup too 😄https://github.com/silverstripe/silverstripe-framework/issues/8498

It's great that users like you care so much about this!
